content/hardware/02.hero/boards/uno-wifi-rev2/tutorials/uno-wifi-rev-2/uno-wifi-r2-chromebook-installation/uno-wifi-r2-chromebook-installation.md

```diff
@@ -0,0 +1,126 @@
+---
+title: 'UNO WiFi Rev 2 Chromebook Setup'
+difficulty: easy
+description: 'A quick tutorial on how to setup your UNO WiFi Rev 2 board with a Chromebook, using the Web Editor & the Arduino Chrome App.'
+tags: 
+  - Chromebook
+  - Installation
+author: 'Karl Söderby'
+hardware:
+  - hardware/03.hero/boards/uno-wifi-rev-2
+software:
+  - web-editor
+---
+
+## Introduction 
+
+> This tutorial is only relevant for Chromebook users that uses an Arduino UNO WiFi Rev 2 board.  
+
+The [UNO WiFi Rev 2](https://store.arduino.cc/arduino-uno-wifi-rev2) is the connected version of the classic UNO board. If you are using a **Chromebook**, setting up your board is a bit different for this particular board:
+
+- You will need to upgrade the firmware using either a **Windows/Mac/Linux** computer, prior to programming it via a Chromebook. Detailed instructions are provided in this tutorial, and the process only takes a few minutes!
+- It is only possible to use the [Web Editor](https://create.arduino.cc/editor), an online IDE that is part of the [Arduino Cloud](https://cloud.arduino.cc/).
+
+***Note that only the Web Editor is supported in Chromebooks. It is not possible to configure and upload to UNO WiFi Rev 2 boards via the [IoT Cloud](https://create.arduino.cc/iot/things).***
+
+## Goals
+
+The goals of this project are:
+
+- Learn how to upgrade the firmware on your UNO WiFi Rev 2, so it can be used with a Chromebook.
+- Learn how to install the [Arduino Create for Education](https://chrome.google.com/webstore/detail/arduino-create-for-educat/elmgohdonjdampbcgefphnlchgocpaij) app from Chrome Web Store.
+- Learn how to upload a sketch to your board using a Chromebook and the Web Editor.
+
+## Hardware & Software Needed
+
+- [Arduino IDE 2](https://www.arduino.cc/en/software#future-version-of-the-arduino-ide). 
+- [Arduino Web Editor](https://create.arduino.cc/).
+- [Arduino Create for Education](https://chrome.google.com/webstore/detail/arduino-create-for-educat/elmgohdonjdampbcgefphnlchgocpaij) (Chrome Web Store)
+- [Arduino UNO WiFi Rev 2](https://store.arduino.cc/arduino-uno-wifi-rev2).
+
+***Please note that you need to install the Arduino IDE 2.0.X. The legacy IDE (1.8.X) will not work for this tutorial.***
+
+## Upgrading Firmware
+
+Since Chromebooks cannot run executables, the firmware upgrade for the UNO WiFi Rev 2 needs to be done through a computer with the Windows/Mac/Linux OS installed. 
+
+***While you won't be using the Arduino IDE 2.0 directly, you will be utilizing a set of tools that is included in the IDE. This is why you need to install the editor on your local machine.***
```

Comment:
Which tools? The only one I see used here is AVRDUDE. That is not included with Arduino IDE 2.x.

It is included with Arduino IDE 1.x, but not at the path shown in the instructions. And it would be more efficient to just download AVRDUDE directly.

Reply:
@per1234 I have `avrdude` installed at that exact path, and I was thinking it was there because of the IDE 2, no?
I don't remember I ever installed the version 1.8 before (but I may remember wrong).

Reply:
It is installed there by Arduino IDE 2.x on the first run of a fresh installation. It just automatically uses Boards Manager to install the "Arduino AVR Boards" platform (https://github.com/arduino/arduino-ide/pull/497). That boards platform installation includes AVRDUDE.

The tutorial doesn't instruct the user to run the IDE.

Reply:
AVRDUDE is bundled with Arduino IDE 1.x in the `hardware/tools/avr/bin` subfolder.

Since this path does not include the AVRDUDE version like the Boards Manager-installed AVRDUDE, it avoids the problem of the path to AVRDUDE changing whenever a new version of the "Arduino AVR Boards" platform is released with a different AVRDUDE tool dependency (there is already an AVRDUDE 6.4 release available, and 7.0 is coming soon).

Download URLs for Arduino's AVRDUDE are in this file:
https://downloads.arduino.cc/packages/package_index.json

The current ones in use by the "Arduino megaAVR Boards" platform are in this object:

```json
{
  "name": "avrdude",
  "version": "6.3.0-arduino17",
  "systems": [
    {
      "size": "219631",
      "checksum": "SHA-256:2a8e68c5d803aa6f902ef219f177ec3a4c28275d85cbe272962ad2cd374f50d1",
      "host": "arm-linux-gnueabihf",
      "archiveFileName": "avrdude-6.3.0-arduino17-armhf-pc-linux-gnu.tar.bz2",
      "url": "http://downloads.arduino.cc/tools/avrdude-6.3.0-arduino17-armhf-pc-linux-gnu.tar.bz2"
    },
    {
      "size": "229852",
      "checksum": "SHA-256:6cf948f751acfe7b96684537f2291c766ec8b54b4f7dc95539864821456fa9fc",
      "host": "aarch64-linux-gnu",
      "archiveFileName": "avrdude-6.3.0-arduino17-aarch64-pc-linux-gnu.tar.bz2",
      "url": "http://downloads.arduino.cc/tools/avrdude-6.3.0-arduino17-aarch64-pc-linux-gnu.tar.bz2"
    },
    {
      "size": "279045",
      "checksum": "SHA-256:120cc9edaae699e7e9ac50b1b8eb0e7d51fdfa555bac54233c2511e6ee5418c9",
      "host": "x86_64-apple-darwin12",
      "archiveFileName": "avrdude-6.3.0-arduino17-x86_64-apple-darwin12.tar.bz2",
      "url": "http://downloads.arduino.cc/tools/avrdude-6.3.0-arduino17-x86_64-apple-darwin12.tar.bz2"
    },
    {
      "size": "254271",
      "checksum": "SHA-256:accdfb920af2aabf4f7461d2ac73c0751760f525216dc4e7657427a78c60d13d",
      "host": "x86_64-linux-gnu",
      "archiveFileName": "avrdude-6.3.0-arduino17-x86_64-pc-linux-gnu.tar.bz2",
      "url": "http://downloads.arduino.cc/tools/avrdude-6.3.0-arduino17-x86_64-pc-linux-gnu.tar.bz2"
    },
    {
      "size": "244550",
      "checksum": "SHA-256:5c8cc6c17db9300e1451fe41cd7178b0442b4490ee6fdbc0aed9811aef96c05f",
      "host": "i686-linux-gnu",
      "archiveFileName": "avrdude-6.3.0-arduino17-i686-pc-linux-gnu.tar.bz2",
      "url": "http://downloads.arduino.cc/tools/avrdude-6.3.0-arduino17-i686-pc-linux-gnu.tar.bz2"
    },
    {
      "size": "328460",
      "checksum": "SHA-256:e99188873c7c5ad8f8f906f068c33600e758b2e36cce3adbd518a21bd266749d",
      "host": "i686-mingw32",
      "archiveFileName": "avrdude-6.3.0-arduino17-i686-w64-mingw32.zip",
      "url": "http://downloads.arduino.cc/tools/avrdude-6.3.0-arduino17-i686-w64-mingw32.zip"
    }
  ]
}
```